src/topics/badge/badge.ts

```diff
@@ -1,5 +1,6 @@
 import { BigNumber, BigNumberish, ethers } from "ethers";
 import { Attester } from "../attester";
+import { ConstructedBadge } from "./types";
 
 export type BadgeConstructor = {
   name?: string;
```

Comment:
name should not be mandatory ?

Below, in Badge constructor, `requirements` and `collectionId` are not used